src/login/lobby.cpp

```diff
@@ -198,10 +198,12 @@ int32 lobbydata_parse(int32 fd)
                         uint8  charIdExtra = (charId >> 16) & 0xFF;
 
                         // uList is sent through data socket (to bootloader)
-                        ref<uint32>(uList, 16 * (i + 1)) = contentId;
-                        ref<uint16>(uList, 20 * (i + 1)) = charIdMain;
-                        ref<uint8>(uList, 22 * (i + 1))  = worldId;
-                        ref<uint8>(uList, 23 * (i + 1))  = charIdExtra;
+                        uint32 uListOffset = 16 * (i + 1);
+
+                        ref<uint32>(uList, uListOffset)     = contentId;
+                        ref<uint16>(uList, uListOffset + 4) = charIdMain;
```

Comment:
is the math change intended?
For the second line:
`20 * (i + 1)` != `16 * (i + 1) + 4`
or is it equal and I suck at math? (very possible)

Reply:
Yes. the offsets were _very_ wrong. They were supposed to be packed next to eachother, not dramatically away.